### PR TITLE
feat(nimbus): Support multiple FML paths per app; fix focus-ios fetching

### LINKED ISSUE #9804: focus-ios FML fetching is broken on main
The Focus iOS and Firefox iOS are merging into a monorepo. In preparation for that, [https://github.com/mozilla-mobile/focus-ios/pull/3917](https://github.com/mozilla-mobile/focus-ios/pull/3917|smart-link)  landed in focus-ios earlier today which moves all the focus-specific stuff, including its FML, into a subdirectory. This breaks our fetch logic, which will need to be updating to support fetching different paths in different branches.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4063)

### PR DESCRIPTION
**Pull request by brennie**
Because

- focus-ios had its nimbus.fml.yaml file move recently (see https://github.com/mozilla-mobile/focus-ios/pull/3917)

This commit

- updates our fetch logic to support multiple potential FML file locations; and
- updates focus_ios.fml_path to support the old and new paths.

Fixes #9804

### CODE REVIEW

File: experimenter/manifesttool/fetch.py [hunk 1]
```diff
@@ -67,7 +67,18 @@ def fetch_fml_app(
         else:
             print(f"fetch: {app_name} at {ref}")
 
-        channels = nimbus_cli.get_channels(app_config, ref.target)
+        if isinstance(app_config.fml_path, str):
+            fml_path = fml_path
```
Comment (jaredlockhart) [inline experimenter/manifesttool/fetch.py:71]: Bwa?

File: experimenter/manifesttool/fetch.py [hunk 2]
```diff
@@ -67,7 +67,18 @@ def fetch_fml_app(
...
+        elif isinstance(app_config.fml_path, list):
+            for fml_path in app_config.fml_path:
+                if github_api.file_exists(app_config.repo.name, fml_path, ref.target):
+                    break
```
[the remark this answers is not in the document]
Comment (brennie) [reply experimenter/manifesttool/fetch.py:75]: Yeah we expect there only to be one valid top level FML per app

File: experimenter/manifesttool/appconfig.py [hunk 3]
```diff
@@ -128,7 +128,7 @@ class AppConfig(BaseModel):
 
     slug: str
     repo: Repository
-    fml_path: Optional[str]
+    fml_path: Optional[Union[str, list[str]]]
```
[the remark this answers is not in the document]
Comment (brennie) [reply experimenter/manifesttool/appconfig.py:131]: No because this should be a special case.

**Review by jaredlockhart — APPROVED** [review]
Coolio looks good, good tests, wonder if we'll have to use this pattern again as more repos move around, good to set up a mechanism for it now 👍

**Comment by brennie** [comment]
> Coolio looks good, good tests, wonder if we'll have to use this pattern again as more repos move around, good to set up a mechanism for it now 👍

firefox-ios is going to be moving into a sub dir as well, so we'll have to use it for that, too.

**Pull request merged**